internal/gir/types/template.go

```diff
@@ -70,6 +78,47 @@ func (f *funcArgsTemplate) AddAPI(t string, n string, k Kind, ns string) {
 	f.API.Full = append(f.API.Full, n+" "+t)
 }
 
+func (f *funcArgsTemplate) AddPtrCallbacks(t string, n string, k Kind, ns string) {
```

Comment:
I think it's better that we make these functions a receiver on argsTemplate instead and avoid this duplicated code (would need a boolean passed though so that we can different things in the callbacktype :/), but maybe that is something for me to handle later

Reply:
I started with that approach, but because the arg types are different it didn't work out and this was just simpler to reason about, even though I was unhappy about all the duplication.

Reply:
These Ptr functions are no longer included in this PR.